src/line.js

```diff
@@ -25,16 +29,19 @@ define([
   }
 
   Line.prototype.lineLength = function() {
-    var self = this;
     var length = 0;
 
-    this._sources.forEach(function(sourceGroup) {
-      var sourceGroupLength = self._view.timeToPixels(sourceGroup._source.endTime);
+    for (var sourceId in this._sources) {
+      if (Utils.objectHasProperty(this._sources, sourceId)) {
+        var sourceGroupLength = this._view.timeToPixels(
+          this._sources[sourceId].sourceGroup.getSource().endTime
```

Comment:
We can create a method like:
```
this._sources[sourceId].getSource()
```
which will be exactly this:
```
this.sourceGroup.getSource()
```

Reply:
Ok

Reply:
We can't do that without creating a new class right? I am not sure it's worth it if we don't go for a DoublyLinkedList class

Reply:
I thought we already had a class for "this._sources[sourceId]" and we could add to it a new method.

Reply:
ah, ok, then let's keep it like this.